### PR TITLE
Add check for HEAD ref in diagnostics

### PR DESCRIPTION
**Pull request by BenMusch**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Addresses this issue: https://github.com/Homebrew/brew/issues/3068 by extending a diagnostic check so that it adds a warning to the console if `.git/HEAD` is not set to `/refs/heads/master`

Any thoughts on how to test this? Are there patterns established for testing things that interact with the filesystem?

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -785,6 +785,22 @@ def check_coretap_git_origin
               git -C "#{coretap_path}" remote set-url origin #{Formatter.url("https://github.com/Homebrew/homebrew-core.git")}
           EOS
         end
+
+        head = coretap_path.git_head
+        return unless head && head !~ %r{refs/heads/master}
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb]: Don't use `unless  ... &&`; it's hard to follow. Switch to an `if`.

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -785,6 +785,22 @@ def check_coretap_git_origin
...
+          Suspicious #{CoreTap.instance} git head found.
+
+          With a non-standard head, your local version of Homebrew might not
+          have all of the changes intended for the most recent release. The
+          current git head is:
+            #{head}
+
+          Unless you have compelling reasons, consider setting the head to
+          point at the master branch by running:
+            git checkout master
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb]: Replace all the above text with something like:

```
Homebrew/homebrew-core is not on the master branch

Check out the master branch by running:
  git -C "$(brew --repo homebrew/core)" checkout master
```

Want to avoid Git jargon where possible here.

**Comment by MikeMcQuaid** [comment]
Fantastic work again, @BenMusch! Looking forward already to your next PR 😍

**Comment by MikeMcQuaid** [comment]
@benmusch this had to be reverted unfortunately (see the linked stuff above) but I’d love if you could try again in a new PR and detail what commands you use to test this so we can try them too. Thanks again!

**Comment by BenMusch** [comment]
Interesting, will investigate and re-open. Thanks for the heads up @MikeMcQuaid

**Comment by ilovezfs** [comment]
Thanks, @BenMusch! For what it's worth, I think you need to use `--symbolic-full-name`.
